### PR TITLE
Run the suite under musl, and the frameworks under the safety checks

### PR DESCRIPTION
**Pull request by Kludex**
The other half of running the framework suites where the wheels are published; [#49](https://github.com/Kludex/zuvloop/pull/49) is the tests, this is the platforms.

**musl has only ever been compiled for, never run.** It is a third of the published wheels and a different libc under the same Zig `std.c` shapes, which is exactly the class of defect that keeps reaching users ahead of CI - four have shipped that a Mac could not see. A new job runs the whole suite under Alpine.

Two things about how it is wired, both deliberate:

- Alpine runs in a plain `docker run` rather than as a job-level `container:`. The Actions runner needs a glibc Node to start a container and Alpine has none, so a `container:` job would fail before reaching the tests.
- Zig is fetched as its own static tarball, and uv is pinned to the image's interpreter with `UV_PYTHON_PREFERENCE=only-system` - uv's managed Python builds are linked against glibc, so letting it choose would quietly test the wrong libc.

**`release-safe` now installs the bench group.** It syncs `dev` alone today, so it could not import `tests/test_frameworks.py` at all - which is what broke [#49](https://github.com/Kludex/zuvloop/pull/49) on first push. That PR makes the module skip when the imports are missing, so this is the difference between skipping the framework tests there and running them: real HTTP traffic with every bounds and overflow check left on, which is the most valuable place to run them.

I cannot verify the musl job locally - the Docker daemon on this machine would not start - so it may want a round of iteration in CI. The two things I could check are that the Zig tarball URL resolves (the older `zig-linux-x86_64-…` naming is a 404) and that `python:3.14-alpine` exists.

## AI Disclaimer

This PR was developed with the assistance of either Claude or Codex. I've reviewed and verified the changes.

---
## Summary by cubic
Run the full test suite on musl (Alpine) and enable framework tests under safety checks. This validates musl wheels at runtime, exercises real HTTP paths, and reduces CI supply-chain risk.

- Add a musl CI job via `docker run` on `python:3.14-alpine`; install `openssl`; fetch Zig 0.16.0 with a pinned SHA-256; use `uv` with `UV_PYTHON_PREFERENCE=only-system`; run the suite inside the container rather than a job-level `container:`.
- Pin `actions/checkout@v7.0.1` with `persist-credentials: false` to avoid leaking the repo token into the container.
- Update `release-safe` to install the `bench` group so framework tests (`aiohttp`, `uvicorn`) run with bounds/overflow checks on.
- Loosen DNS test to accept libc-specific canonical name behavior (musl fills it); keep the standard library parity check.

<sup>Written for commit 2059de8eef4f0ff1a547bca2e5f7f31fcef00c0c. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Kludex/zuvloop/pull/50?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Tests**
  * Added continuous integration coverage for musl-based environments using Python 3.14 Alpine.
  * Added validation with Zig and required native development dependencies.

* **Chores**
  * Updated release verification to include benchmark dependencies.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -65,6 +65,40 @@ jobs:
...
+  # of defect that keeps reaching users is exactly the one a host cannot see.
+  #
+  # Alpine runs in a plain container rather than as a job `container:`, because
+  # the Actions runner needs a glibc Node to start one and Alpine has none. Zig
+  # is fetched as its own static tarball for the same reason.
+  musl:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v7
```
[the remark this answers is not in the document]
Comment (Kludex) [reply .github/workflows/ci.yml]: Applied in c92fbb2. It matters here specifically because this job mounts the workspace into a container that then runs apk and package code from the index, so a persisted token would be sitting in .git/config where that code could read it. Nothing in the job pushes.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -65,6 +65,40 @@ jobs:
...
+
+      - name: Test under musl
+        run: |
+          docker run --rm -v "$PWD:/src" -w /src python:3.14-alpine sh -euc '
+            apk add --no-cache curl xz build-base
+
+            curl -fsSL https://ziglang.org/download/0.16.0/zig-x86_64-linux-0.16.0.tar.xz \
+              | tar -xJ -C /opt
```
[the remark this answers is not in the document]
Comment (Kludex) [reply .github/workflows/ci.yml]: Applied in c92fbb2, using the SHA256 that ziglang.org publishes rather than minisign. Same guarantee for this purpose and it pins the exact artifact rather than any signed one, without needing a minisign package and a pinned public key inside Alpine. Checked by downloading the real tarball and running sha256sum -c against the pin.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -65,6 +65,40 @@ jobs:
...
+            export PATH="/opt/zig-x86_64-linux-0.16.0:$PATH"
+            zig version
+
+            pip install --no-cache-dir --quiet uv
+            # The interpreter in the image is the one to test against; uv would
+            # otherwise fetch a managed build, and those are linked against glibc.
+            export UV_PYTHON_PREFERENCE=only-system
+            uv sync --python "$(command -v python3)" --group dev --group bench
+
+            uv run python -c "import zuvloop; print(zuvloop.libuv_version())"
```
[the remark this answers is not in the document]
Comment (Kludex) [reply .github/workflows/ci.yml:124]: Leaving this one. The step exists to fail fast if the extension does not import or link under musl, and it does that already - an import error exits non-zero and names the problem before 370 tests fail for the same reason. The count(".") == 2 contract it points at is asserted in the suite that runs seconds later, so asserting it here would duplicate a test rather than cover anything new.
